### PR TITLE
chore(curriculum) add step 42 to learning forms by creating registration form

### LINKED ISSUE #46003: Order for CSS Pesudoselector not
### Describe the Issue

It is introduced here:

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-typography-by-building-a-nutrition-label/step-48

But it is used prior to the introduction here:

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-41

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-41

### Your code

N/A

### Expected behavior

Material should be introduced before used.

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46003

this is a remake of #46542 because of merge conflict, @lasjorg I will try to fix the space indenting issue, that the whole the file has, after this gets pushed. 

<details> 
 <summary> step 41</summary>
 
image

</details>

<details> 
 <summary> step 42</summary>
 
image

</details>

<details> 
 <summary> step 43</summary>
 
image

</details>
<details> 
 <summary> step 44</summary>
 
image

</details>
<details> 
 <summary> step 45</summary>
 
 did I find a bug removed `width: 100%` will still make the test pass 
 
image

</details>
<details> 
 <summary> step 46</summary>
 
image

</details>
<details> 
 <summary> step 47</summary>
 
image

</details>

<details> 
 <summary> step 48</summary>
 
image

</details>

<details> 
 <summary> step 49</summary>
 
image

</details>

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
I have changed the order of step 46, because the order of `margin` is as follow

```css
/* top | right | bottom | left */
margin: 2px 1em 0 auto;
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fc22d1e64d1b04cdd4e602.md [hunk 1]
```diff
@@ -11,22 +11,22 @@ Add some space between the `.inline` elements and the `label` text, by giving a
 
 # --hints--
 
-You should give the `.inline` elements a `margin-right` of `0.5em`.
+You should give the `.inline` elements a `margin-top` of `0`.
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fc22d1e64d1b04cdd4e602.md:14]: are these changes for this PR?

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fc22d1e64d1b04cdd4e602.md (hunk 1 above)
Comment (Sboonny) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fc22d1e64d1b04cdd4e602.md:14]: The margin one are quality of life change, the none asked for it and it is unnecessary. I am happy to revert it if asked.

**Review by majestic-owl448 — DISMISSED** [review]

**Comment by lasjorg** [comment]
@Sboonny What is the plan for the tabs/spaces format issue? I still see it in the markdown files.

BTW, if you want to see the whitespace in the code editor you can add the visual aid.

Settings > Editor > Render Whitespace > All

An `->` is a tab and `..` is two spaces.

---

@ojeytonwilliams How did you fix the whitespace issue in the other PR, did you just resave all the files?

What is the reason for `.prettierignore` excluding the challenge files? Does it cause some issues? It would be nice if it could format the files for us to avoid inconsistencies.

**Comment by Sboonny** [comment]
@lasjorg the last time, I did fix it but that created merge conflict too big, that it lead to 2000 files changes. I am waiting for this to pushed, then I will create separate pull request, because the tab change require changing 48 files.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62b30924c5e4ef0daba23b5e.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+You can use the `:last-of-type` pseudo-class to select only the last element of a certain type.
+
+```js
+assert.exists(new __helpers.CSSHelp(document).getStyle('fieldset:last-of-type'));
+```
+
+Your `fieldset:last-of-type` should have `border-bottom` set as `none`.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('fieldset:last-of-type')?.borderBottom, 'none');
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62b30924c5e4ef0daba23b5e.md:29]:
```suggestion
const borderBottom = new __helpers.CSSHelp(document).getStyle('fieldset:last-of-type')?.borderBottom;
assert(borderBottom === 'none' || borderBottom === 'medium none');
```


This test isn't passing in firefox - works in safari and chrome. Firefox gives a value of `medium none` - chrome and safari both give `none`. I made a suggestion that works - feel free to improve it if you want.

**Review by moT01 — COMMENTED** [review]
Found a few minor things here.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 (assuming tests pass)

Edit: Side thought on the tabs in the files. They were pre-existing so I think they're fine for another PR.

**Review by majestic-owl448 — APPROVED** [review]

**Pull request merged**